python-stdlib/logging/logging.py

```diff
@@ -117,12 +162,71 @@ def debug(msg, *args):
     getLogger().debug(msg, *args)
 
 
-def basicConfig(level=INFO, filename=None, stream=None, format=None):
-    global _level, _stream
-    _level = level
-    if stream:
-        _stream = stream
-    if filename is not None:
-        print("logging.basicConfig: filename arg is not supported")
-    if format is not None:
-        print("logging.basicConfig: format arg is not supported")
+def warning(msg, *args):
+    getLogger().warning(msg, *args)
+
+
+def error(msg, *args):
+    getLogger().error(msg, *args)
+
+
+def critical(msg, *args):
+    getLogger().critical(msg, *args)
+
+
+def exception(msg, *args):
+    getLogger().exception(msg, *args)
+
+
+def shutdown():
+    for k, logger in _loggers.items():
+        for h in logger.handlers:
+            h.close()
+        _loggers.pop(logger, None)
+
+
+if hasattr(sys, "atexit"):
+    sys.atexit(shutdown)
+
+
+def addLevelName(level, name):
+    _level_dict[level] = name
+
+
+def log(level, msg, *args):
```

Comment:
Could you put this one and `getLogger` higher up in the file, in the order they are used? That's easier to read. And then the `if hasattr(sys, "atexit")...` part can do towards the end insetad of in the middle of function definitions.

Reply:
Those used to be in `__init__.py` which was much more readable IMHO, I don't mind rearranging anything, but @dpgeorge wants the diff to be minimal.

Reply:
Not really following about the `__init__.py` (I'm looking at latest master 0051a5ef which doesn't have that), but: this is one chunk of new additions, things can be reordered within that chunk?

Reply:
I was actually referring to `atexit` stuff, but also could have moved some those functions to `__init__.py`, it was there but removed. Anyway I don't think the diff is very minimal anyway, so reordered them.

Reply:
I think the global `debug` and `info` can still be swapped, to match their order elsewhere.